src/states/approved_peer/mod.rs

```diff
@@ -1161,34 +1858,110 @@ impl Elder {
         list
     }
 
-    /// Returns a list of target IDs for a message sent via route.
-    fn get_targets(&self, dst: &DstLocation) -> Result<(Vec<P2pNode>, usize), RoutingError> {
-        let (targets, dg_size) = self.chain.targets(dst)?;
-        Ok((targets.into_iter().cloned().collect(), dg_size))
+    // Send message over the network.
+    fn send_signed_message(&mut self, msg: &MessageWithBytes) -> Result<()> {
+        let (targets, dg_size) = self.chain.targets(msg.message_dst())?;
+
+        trace!("{}: Sending {:?} via targets {:?}", self, msg, targets);
+
+        let targets: Vec<_> = targets
+            .into_iter()
+            .filter(|p2p_node| {
+                self.msg_filter
+                    .filter_outgoing(msg, p2p_node.public_id())
+                    .is_new()
+            })
+            .map(|node| *node.peer_addr())
+            .collect();
+
+        let cheap_bytes_clone = msg.full_bytes().clone();
+        self.send_message_to_targets(&targets, dg_size, cheap_bytes_clone);
+
+        Ok(())
     }
 
-    // Signs and proves the given `RoutingMessage` and wraps it in `SignedRoutingMessage`.
-    fn to_accumulating_message(
-        &self,
-        dst: DstLocation,
-        variant: Variant,
-        node_knowledge_override: Option<u64>,
-    ) -> Result<AccumulatingMessage> {
-        let proof = self.chain.prove(&dst, node_knowledge_override);
-        let pk_set = self.our_section_bls_keys().clone();
-        let secret_key = self.chain.our_section_bls_secret_key_share()?;
+    ////////////////////////////////////////////////////////////////////////////
+    // Miscellaneous
+    ////////////////////////////////////////////////////////////////////////////
 
-        let content = PlainMessage {
-            src: *self.our_prefix(),
-            dst,
-            variant,
-        };
+    fn print_rt_size(&self) {
```

Comment:
actually prints more?

Reply:
Yeah, there is more that could be improved in this function, but it's probably out of scope for this PR.

Reply:
I am just suggesting to change the function name  :)